src/components/TodoApp.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useContext } from 'react';
+import { TodosContext, SetTodosContext } from '../contexts/TodosContext';
+import { Todo } from '../types/Todo';
+import { TodoList } from './TodoList';
+import { TodosFilter } from './TodosFilter';
+import { FilterStatus } from '../types/FilterStatus';
+
+export const TodoApp = () => {
+  const [todoTitle, setTodoTitle] = useState<string>('');
+  const [
+    filterStatus, setFilterStatus,
+  ] = useState<FilterStatus>(FilterStatus.All);
+
+  const todosPrep = useContext(TodosContext);
+  const setTodos = useContext(SetTodosContext);
+
+  const todos = todosPrep.filter((todo) => {
+    switch (filterStatus) {
+      case FilterStatus.Active:
+        return !todo.completed;
+      case FilterStatus.Completed:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  const leftTodos = todos.filter((todo) => !todo.completed);
+  const compleatedTodos = todos.filter((todo) => todo.completed);
+  const leftTodosText = leftTodos.length === 1 ? 'item' : 'items';
+
+  const [allCompleted, setAllCompleted] = useState<boolean>(
+    leftTodos.length === 0,
+  );
+
+  React.useEffect(() => {
```

Comment:
import `useEffect` and use directly
```suggestion
  useEffect(() => {
```